### PR TITLE
Implement YSweetConnection class in SDK

### PR DESCRIPTION
**Pull request by vkameswaran**
This PR:

• Adds a DocConnection class in the SDK for making REST requests to documents via ClientToken
• Refactors HTTP concerns into HttpClient class
• Updates DocumentManager methods to work via DocConnection for proper routing in multi-node setups
• Modifies auth to allow server tokens to access any document
• Adds test coverage

Design doc: https://docs.google.com/document/d/10O4-i9sxUL5-bWjIYqgxbStJGqNCVLfP411J9x-fTFM/edit

### CODE REVIEW

File: crates/y-sweet-core/src/auth.rs [hunk 1]
```diff
@@ -323,7 +323,7 @@ impl Authenticator {
                     Err(AuthError::InvalidResource)
                 }
             }
-            _ => Err(AuthError::InvalidResource),
+            Permission::Server => Ok(()), // Server tokens can access any doc.
```
[the remark this answers is not in the document]
Comment (vkameswaran) [reply crates/y-sweet-core/src/auth.rs:326]: @greptileai what kind of vulnerabilities could this introduce?

File: js-pkg/sdk/src/connection.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+   * let doc = new Y.Doc()
+   * doc.transact(() => {
+   *  Y.applyUpdate(doc, update)
+   * })
+   * ```
+   *
+   * @returns
+   */
+  public async getAsUpdate(): Promise<Uint8Array> {
+    const result = await this.client.request(`as-update`, 'GET')
```
[the remark this answers is not in the document]
Comment (vkameswaran) [reply js-pkg/sdk/src/connection.ts:31]: Suggest a code change @greptileai?

[the remark this answers is not in the document]
Comment (vkameswaran) [reply js-pkg/sdk/src/connection.ts:31]: @greptileai suggest a code change?

[the remark this answers is not in the document]
Comment (vkameswaran) [reply js-pkg/sdk/src/connection.ts:31]: @greptileai suggest a change?

File: js-pkg/sdk/src/http.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  public async request(url: string, method: 'GET'): Promise<Response>
+  public async request(url: string, method: 'POST', body: BodyType): Promise<Response>
+
+  public async request(path: string, method: 'GET' | 'POST', body?: BodyType): Promise<Response> {
+    const headers = new Headers()
+    if (this.token) {
+      headers.set('Authorization', `Bearer ${this.token}`)
+    }
+
+    let rawBody: string | Uint8Array
```
[the remark this answers is not in the document]
Comment (vkameswaran) [reply js-pkg/sdk/src/http.ts:29]: Write a suggestion for this in ``` syntax @greptileai

[the remark this answers is not in the document]
Comment (vkameswaran) [reply js-pkg/sdk/src/http.ts:29]: @greptileai write a suggestion please

File: js-pkg/sdk/src/http.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      headers.set('Content-Type', 'application/json')
+      rawBody = JSON.stringify(body)
+    }
+
+    // NOTE: In some environments (e.g. NextJS), responses are cached by default. Disabling
+    // the cache using `cache: 'no-store'` causes fetch() to error in other environments
+    // (e.g. Cloudflare Workers). To work around this, we simply add a cache-busting query
+    // param.
+    const cacheBust = generateRandomString()
+    let url = `${this.baseUrl}/${path}?z=${cacheBust}`
```
[the remark this answers is not in the document]
Comment (rajitkhanna) [reply js-pkg/sdk/src/http.ts:43]: Can you help me here @greptileai

**Pull request closed without merging**